source/draw/README.md

```diff
@@ -0,0 +1,27 @@
+# Draw API
+
+## Image
+An image is responsible for storing an array of pixels in 2 or 3 dimensions on CPU or GPU.
+
+## Canvas
+An abstraction layer for drawing that is responsible for portability between underlaying versions of OpenGL using class inheritance.
+
+## Target
+The target is the image being drawn to. The same image cannot be both input and output to the same draw call because of hardware limitations on some graphics cards.
+
+## Viewport
+The viewport is the sub region of the target image that is use whole image by default. This can be set to a smaller region to store multiple images in an atlas.
+
+No pixels will be drawn outside of the viewport. If you apply a transform, it will be centered to the viewport and any pixel leaving the area will be clipped by the viewport.
+
+## Destination
+The destination is similar to the viewport by defining where to draw. The difference is how they work together with the transform. Viewport is deciding the center of the transform and where to clip so with a default viewport, the transform will be centered around the whole target image instead of being centered around the destination. The destination may leave its original location using a transform. The destination is just a way to shrink the drawn quad shape.
```

Comment:
I think this sentence could be splitted into two
`Viewport is deciding the center of the transform and where to clip so with a default viewport, the transform will be centered around the whole target image instead of being centered around the destination.`
or something to make it more understandable. @marcusnaslund ?